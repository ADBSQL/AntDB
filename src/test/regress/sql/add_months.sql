set grammar to oracle;
set datestyle='ISO,YMD';
create table addmonthstbl(id int, dt timestamp);
insert into addmonthstbl values(1,to_date('2000-3-31','YYYY-MM-DD'));
insert into addmonthstbl values(2,to_date('2000-2-29','YYYY-MM-DD'));
select add_months(to_date('2000-3-31','YYYY-MM-DD'),1) from dual; 
select id, add_months(dt,1) from addmonthstbl where id=1;
select add_months(to_date('2000-3-31','YYYY-MM-DD'),-1) from dual; 
select id, add_months(dt,1) from addmonthstbl where id=1;
select add_months(to_date('2000-3-31','YYYY-MM-DD'),0) from dual; 
select id, add_months(dt,0) from addmonthstbl where id=1;
select add_months(to_date('2000-2-29','YYYY-MM-DD'),12) from dual; 
select id, add_months(dt,12) from addmonthstbl where id=2;
select add_months(to_date('2000-2-29','YYYY-MM-DD'),-12) from dual; 
select id, add_months(dt,-12) from addmonthstbl where id=2;
select add_months(to_date('2000-2-29','YYYY-MM-DD'),12.1) from dual; 
select add_months(to_date('2000-2-29','YYYY-MM-DD'),-12.55) from dual; 
select add_months(to_date('1-2-28','YYYY-MM-DD'),-300) from dual;
select add_months(to_date('6000-2-29','YYYY-MM-DD'),30000) from dual;
select add_months(to_date('2000-2-29','YYYY-MM-DD'),'3') from dual;
select add_months(to_date('2000-2-29','YYYY-MM-DD'),'') from dual;
select add_months(to_date('2000-2-29','YYYY-MM-DD'),null) from dual;
select add_months(null,null) from dual;
select add_months('','') from dual;
select add_months('',null) from dual;
select add_months(null,'') from dual;
select add_months('',3) from dual;
drop table addmonthstbl;
