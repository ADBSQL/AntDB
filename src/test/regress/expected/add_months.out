set grammar to oracle;
set datestyle='ISO,YMD';
create table addmonthstbl(id int, dt timestamp);
insert into addmonthstbl values(1,to_date('2000-3-31','YYYY-MM-DD'));
insert into addmonthstbl values(2,to_date('2000-2-29','YYYY-MM-DD'));
select add_months(to_date('2000-3-31','YYYY-MM-DD'),1) from dual; 
 ADD_MONTHS(TO_DATE('2000-3-31','YYYY-MM-DD'),1)  
--------------------------------------------------
 2000-04-30 00:00:00
(1 row)

select id, add_months(dt,1) from addmonthstbl where id=1;
 ID |  ADD_MONTHS(DT,1)   
----+---------------------
  1 | 2000-04-30 00:00:00
(1 row)

select add_months(to_date('2000-3-31','YYYY-MM-DD'),-1) from dual; 
 ADD_MONTHS(TO_DATE('2000-3-31','YYYY-MM-DD'),-1)  
---------------------------------------------------
 2000-02-29 00:00:00
(1 row)

select id, add_months(dt,1) from addmonthstbl where id=1;
 ID |  ADD_MONTHS(DT,1)   
----+---------------------
  1 | 2000-04-30 00:00:00
(1 row)

select add_months(to_date('2000-3-31','YYYY-MM-DD'),0) from dual; 
 ADD_MONTHS(TO_DATE('2000-3-31','YYYY-MM-DD'),0)  
--------------------------------------------------
 2000-03-31 00:00:00
(1 row)

select id, add_months(dt,0) from addmonthstbl where id=1;
 ID |  ADD_MONTHS(DT,0)   
----+---------------------
  1 | 2000-03-31 00:00:00
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),12) from dual; 
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),12)  
---------------------------------------------------
 2001-02-28 00:00:00
(1 row)

select id, add_months(dt,12) from addmonthstbl where id=2;
 ID | ADD_MONTHS(DT,12)   
----+---------------------
  2 | 2001-02-28 00:00:00
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),-12) from dual; 
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),-12)  
----------------------------------------------------
 1999-02-28 00:00:00
(1 row)

select id, add_months(dt,-12) from addmonthstbl where id=2;
 ID | ADD_MONTHS(DT,-12)  
----+---------------------
  2 | 1999-02-28 00:00:00
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),12.1) from dual; 
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),12.1)  
-----------------------------------------------------
 2001-02-28 00:00:00
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),-12.55) from dual; 
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),-12.55)  
-------------------------------------------------------
 1999-02-28 00:00:00
(1 row)

select add_months(to_date('1-2-28','YYYY-MM-DD'),-300) from dual;
 ADD_MONTHS(TO_DATE('1-2-28','YYYY-MM-DD'),-300)  
--------------------------------------------------
 0024-02-29 00:00:00
(1 row)

select add_months(to_date('6000-2-29','YYYY-MM-DD'),30000) from dual;
 ADD_MONTHS(TO_DATE('6000-2-29','YYYY-MM-DD'),30000)  
------------------------------------------------------
 8500-02-28 00:00:00
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),'3') from dual;
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),'3')  
----------------------------------------------------
 2000-05-31 00:00:00
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),'') from dual;
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),'')  
---------------------------------------------------
 
(1 row)

select add_months(to_date('2000-2-29','YYYY-MM-DD'),null) from dual;
 ADD_MONTHS(TO_DATE('2000-2-29','YYYY-MM-DD'),NULL)  
-----------------------------------------------------
 
(1 row)

select add_months(null,null) from dual;
 ADD_MONTHS(NULL,NULL)  
------------------------
 
(1 row)

select add_months('','') from dual;
 ADD_MONTHS('','')  
--------------------
 
(1 row)

select add_months('',null) from dual;
 ADD_MONTHS('',NULL)  
----------------------
 
(1 row)

select add_months(null,'') from dual;
 ADD_MONTHS(NULL,'')  
----------------------
 
(1 row)

select add_months('',3) from dual;
 ADD_MONTHS('',3)  
-------------------
 
(1 row)

drop table addmonthstbl;
